modMcf/src/org/aion/mcf/db/ContractDetailsCacheImpl.java

```diff
@@ -49,23 +50,23 @@ public ContractDetailsCacheImpl(IContractDetails<DataWord> origContract) {
     }
 
     @Override
-    public void put(DataWord key, DataWord value) {
+    public void put(IDataWord key, IDataWord value) {
         storage.put(key, value);
         setDirty(true);
     }
 
     @Override
-    public DataWord get(DataWord key) {
+    public IDataWord get(IDataWord key) {
 
-        DataWord value = storage.get(key);
+        IDataWord value = storage.get(key);
         if (value != null) {
-            value = value.clone();
+            value = value.copy();
         } else {
             if (origContract == null) {
                 return null;
             }
             value = origContract.get(key);
-            storage.put(key.clone(), value == null ? DataWord.ZERO.clone() : value.clone());
+            storage.put(key.copy(), value.isZero() ? DataWord.ZERO.copy() : value.copy());
```

Comment:
if this is called with a zero `DoubleDataWord` won't it return a `DataWord.ZERO`?

Reply:
Since zero arrays are not valid entries there is no difference between single and double zero arrays, all that matters is that it is a zero array, so better to return the shorter version. Again, see my comments above, I may have the wrong idea about these zero byte arrays

Reply:
ok. looks good.